src/components/TodoItem.tsx

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (id: number) => Promise<unknown>,
+  updatedTodo: (newTodo: Todo) => Promise<unknown>,
```

Comment:
```suggestion
  updateTodo: (newTodo: Todo) => Promise<Todo>,
```
it returns todo (you may check it in the network tab in devTools also)
update**d**Todo sounds like it's an object of todo that was updated, function should contain verb in the name